pylint/lint.py

```diff
@@ -946,12 +946,20 @@ def register_checker(self, checker):
 
         :raises ValueError: If the priority of the checker is invalid.
         """
-        if checker.name in self._checkers:
-            # TODO: Raise if classes are the same
-            for duplicate in self._checkers[checker.name]:
-                msg = 'A checker called {} has already been registered ({}).'
-                msg = msg.format(checker.name, duplicate.__class__)
-                warnings.warn(msg)
+        existing_checker_types = set(
+            type(existing_checker)
+            for name_checkers in self._checkers.values()
+            for existing_checker in name_checkers
+        )
+        checker_type = type(checker)
+        if checker_type in existing_checker_types:
```

Comment:
Not using isinstance becasue I didn't think it was unreasonable for one checker to subclass another. Wouldn't want to skip different checkers if that was the case. Can change to isinstance if needed.

Reply:
Leaving this as it is, thanks.